frontend/packages/ceph-storage-plugin/src/utils/create-storage-system.ts

```diff
@@ -26,8 +26,8 @@ export const getSupportedVendors = (csv: ClusterServiceVersionKind): string[] =>
 export const getStorageSystemKind = ({ kind, apiVersion, apiGroup }) =>
   `${kind.toLowerCase()}.${apiGroup}/${apiVersion}`;
 
-export const getExternalSubSystemName = (name: string = '') =>
-  name.toLowerCase().replace(/\s/g, '-');
+export const getExternalSubSystemName = (name: string = '', storageClassName: string = '') =>
+  `${name.toLowerCase().replace(/\s/g, '-')}-${storageClassName}`.substring(0, 230);
```

Comment:
why it is exactly 0-230?

Reply:
The max length that k8s support for naming resources is 253.
230 = 253 - (length of suffix added in storage system name)

Reply:
```suggestion
   const scName = storageClassName ? `-${storageClassName}` : '';
   return `${name.toLowerCase().replace(/\s/g, '-')}${scName}`.substring(0, 230);
```
nit: Name's ending on a `-` is weird.

Reply:
storageClassName will always be there. Its not ending in '-'

Reply:
253 - (length of suffix added in storage system name) is suffix length is always 23? But i feel if any one change then entire logic will fail.

Reply:
The length of suffix is 14 - "-storagesystem". This is fixed. Just kept it to 230 to ensure any more changes.
So if  the <sub system resource name> is  always 230 characters .Then the storage system resource name will be 230+14 = 244. Never more than 244 characters. 
We dont need to change this logic and this is very edge case , if at all somehow user is automating to create sub system with such long name ! (may be I overthink here)

Reply:
Can we make it simple like storageClassName.split("-storagesystem")[0]

Reply:
No, its not like that.
This name is used like this here https://github.com/openshift/console/blob/bc6e954c3ff1ce8e6a25dec8a0bbb9b7167cf834/frontend/packages/ceph-storage-plugin/src/components/create-storage-system/payloads.ts#L31

Reply:
So now you see making sure its at least 230 will be a good heuristic